website/addons/twofactor/views.py

```diff
@@ -25,3 +26,37 @@ def user_settings(user_addon, *args, **kwargs):
         message_short='Forbidden',
         message_long='The two-factor verification code you provided is invalid.'
     ))
+
+@must_be_logged_in
+def twofactor_settings_get(auth, *args, **kwargs):
+    return {
+        'result': serialize_settings(auth),
+    }
+
+
+@must_be_logged_in
+def twofactor_enable(auth, *args, **kwargs):
+    user = auth.user
+
+    if user.has_addon('twofactor'):
+        return HTTPError(http.BAD_REQUEST, message='This user already has twofactor enabled')
```

Comment:
The correct way to pass custom error messages is by passing the `data` argument:

```python
HTTPError(http.BAD_REQUEST, data=dict(message_long='This user already has twofactor enabled'))
```